### PR TITLE
net: lwm2m: replace coap_next_block() w/ coap_update_from_block()

### PR DESCRIPTION
**Pull request by rbtchc**
We should call coap_update_from_block() which will determine the minimum
size of the BLOCK1 SIZE between server/client and update the current
offset and total size(if available) accordingly.

This PR shouldn't be conflict w/ #4557

### CODE REVIEW

**Comment by mike-scott** [comment]
@rbtchc This looks good, but I had to do quite a bit of work to my backend changes forhttps://github.com/zephyrproject-rtos/zephyr/pull/4557.  Can we hold this till after that's merged.  I touched some of this code.

**Comment by rbtchc** [comment]
@mike-scott no problem. let's wait.

**Review by mike-scott — CHANGES_REQUESTED** [review]
Holding for now.

**Comment by rbtchc** [comment]
Rebase on top of the master branch which has  #4557 merged now.
@mike-scott plz let me know if you have any problem w/ it.

**Comment by mike-scott** [comment]
LGTM.  Have you run tests to confirm that block size negotiation is working correctly?

**Review by mike-scott — APPROVED** [review]

**Comment by rbtchc** [comment]
@mike-scott Yes, I did. It will now use the minimum block_size between server/client. Originally, it will use whatever we set to "CONFIG_LWM2M_COAP_BLOCK_SIZE"

**Pull request merged**
